music_this_week_app/backend/spotifyHandler.py

```diff
@@ -76,22 +109,39 @@ def filter_artist(self, artist_name):
                     print ("Exact match!")
                 else:
                     print ("Close enough...")
-            return artists[0]['uri']
+            artist_uri = artists[0]['uri']
 
         elif num_matches > 1:
             if VERBOSE:
                 print ("%i matches found: " % num_matches + str([a['name'] for a in artists]) )
             # check for exact match
             for a in artists:
+                no_match_found =True 
```

Comment:
I'm not sure this is correct. I haven't tested, but it stands out as maybe having a couple problems:

- The variable is defined within the scope of the For loop. I'm not sure if it continues to exist after the For loop to check `if no_match_found`
- This part may be correct but is confusing: It is repeatedly set to True on each iteration of the loop, and then set to False followed by a Break if there is a successful match.

I'll change it to set to True once before the For loop.